File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/dispatcher/BaseDispatchResult.java
package io.appium.espressoserver.lib.helpers.w3c.dispatcher;

import java.util.concurrent.Callable;

import javax.annotation.Nullable;

import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;

public abstract class BaseDispatchResult {

    private Callable<BaseDispatchResult> next;
    public boolean hasNext() {
        return next != null;
    }

    public abstract void perform() throws AppiumException;

    @Nullable
    public Callable<BaseDispatchResult> getNext() {
        return next;
    }

    public void setNext(Callable<BaseDispatchResult> next) {
        this.next = next;
    }
}

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/dispatcher/BaseDispatchResult.kt
package io.appium.espressoserver.lib.helpers.w3c.dispatcher

import io.appium.espressoserver.lib.handlers.exceptions.AppiumException
import java.util.concurrent.Callable

abstract class BaseDispatchResult {
    var next: Callable<BaseDispatchResult>? = null
    operator fun hasNext(): Boolean {
        return next != null
    }

    @Throws(AppiumException::class)
    abstract fun perform()
}